### PR TITLE
{bio,data,math}[foss/2021b] Bambi v0.7.1, netcdf4-python v1.5.7, ArviZ v0.11.4 w/ Python 3.9.6, PyMC3 v3.11.1, Theano  v1.1.2

### PR DESCRIPTION
**Pull request by ItIsI-Orient**
(created using `eb --new-pr`)
* https://github.com/easybuilders/easybuild-easyconfigs/pull/14809

### CODE REVIEW

**Comment by ItIsI-Orient** [comment]
Test report by @ItIsI-Orient
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3202.victini.os - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/ee9fe7e959f51aeafc02cca481672cab for a full test report.

**Comment by ItIsI-Orient** [comment]
First merge: https://github.com/easybuilders/easybuild-easyconfigs/pull/14809

**Comment by boegel** [comment]
replaced with:

* #14890
* #14810 
* #14811

**Comment by ItIsI-Orient** [comment]
I have just realized, that this is still relevant, because this PR is for toolchain foss2021b

File: easybuild/easyconfigs/b/Bambi/Bambi-0.7.1-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+to make it extremely easy to fit Bayesian mixed-effects models common in biology,
+social sciences and other disciplines."""
+
+toolchain = {'name': 'foss', 'version': '2021b'}
+
+dependencies = [
+    ('Python', '3.9.6'),
+    ('SciPy-bundle', '2021.10'),
+    ('statsmodels', '0.13.1'),
+    ('ArviZ', '0.11.4'),
```
Comment (boegel) [inline easybuild/easyconfigs/b/Bambi/Bambi-0.7.1-foss-2021b.eb:18]: missing `PyMC3` dependency (same issue as in #14811)

File: easybuild/easyconfigs/b/Bambi/Bambi-0.7.1-foss-2021b.eb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+]
+
+use_pip = True
+sanity_pip_check = True
+
+exts_list = [
+    ('formulae', '0.2.0', {
+        'checksums': ['228335df9907750793016dfb23a8a0cb8d6d9ce9a026bfe2b459aa19f72541d2'],
+    }),
```
Comment (boegel) [inline easybuild/easyconfigs/b/Bambi/Bambi-0.7.1-foss-2021b.eb:27]: You're not actually installing `Bambi` at all here :)

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/t/Theano/Theano-1.1.2-foss-2021b-PyMC.eb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+sanity_check_paths = {
+    'files': ['bin/theano-cache'],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
+
+local_blas_warning = "Using NumPy C-API based implementation for BLAS functions"
+sanity_check_commands = [
+    # make sure that warning about falling back to (slow) NumPy C-API for BLAS functions is *not* printed
+    # mkl-service dependency is required to ensure this doesn't happen
+    "python -c 'import theano.tensor' 2>&1 | grep '%s'; test $? != 0" % local_blas_warning,
```
Comment (boegel) [inline easybuild/easyconfigs/t/Theano/Theano-1.1.2-foss-2021b-PyMC.eb:37]: Hmm, seems like this passes even though `mkl-service` is not included as a dependency (anymore)...
We should check this a bit more carefully.

Does this check in `Theano-1.1.2-intel-2020b-PyMC.eb` fail if `mkl-service` is not used as a dependency?

Please test with

```
eb Theano-1.1.2-intel-2020b-PyMC.eb --filter-deps mkl-serivce --rebuild
```

Comment (boegel) [reply easybuild/easyconfigs/t/Theano/Theano-1.1.2-foss-2021b-PyMC.eb:37]: OK, it seems I was mixing things up a bit here...

`mkl-service` is only need when installing Theano with the `intel` toolchain; if `mkl-service` is removed, this sanity check command fails, because a fallback to the (very slow) NumPy C-API for BLAS fuctions is detected.

When using the `foss` toolchain to install Theano however, `mkl-service` is irrelevant because `MKL` is not a part of the `foss` toolchain. And hence this sanity check passes even when not including `mkl-service` as a dependency. There is no fallback to the slow NumPy C-API for BLAS in this case, since the OpenBLAS component in `foss` is used (and that works fine without the `mkl-service` "trick").

So, long story short: this PR is probably ready to go pretty much, I'll give it another review/test, but this particular comment is not a blocker for this PR at all...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3506.doduo.os - Linux RHEL 8.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/2b6c0fa719e399cbc0ad18f0bea1eafe for a full test report.

File: easybuild/easyconfigs/b/Bambi/Bambi-0.7.1-foss-2021b.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/b/Bambi/Bambi-0.7.1-foss-2021b.eb:28]: fixed in 262d54aaf5534722cd632b32ad736176303215fd

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node2620.swalot.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/b623c7b9dd1f87cfedd506f163c4c141 for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @ItIsI-Orient!
